File: 099--largest-exponential/main.py
import math

def main():
    pairs = read_and_parse()
    largest = 0
    result = max(enumerate(pairs, start=1), key=lambda x: x[1][1] * math.log(x[1][0]))[0]
    print(result)

def read_and_parse():
    lines = (open('base_exp.txt')
        .read()
        .strip()
        .split('\n'))
    return [[int(n) for n in line.split(',')] for line in lines]

main()

File: 099--largest-exponential/main.js
const fs = require('fs');

function main() {
  const pairs = read_and_parse();
  let [result, _] = maxBy(
    pairs.entries(),
    ([i, [base, exp]]) => exp * Math.log(base)
  );
  result += 1;
  console.log(result);
}

function read_and_parse() {
  return fs.readFileSync('base_exp.txt', 'utf8')
    .trim()
    .split('\n')
    .map(line => line.split(',') .map(n => +n));
}

function maxBy(seq, iteratee) {
  return Array.from(seq).reduce((a, b) => iteratee(a) > iteratee(b) ? a : b);
}

main();
